// montydb/varietals.js
'use strict';
module.exports = function(sequelize, DataTypes) {
  var Varietals = sequelize.define('Varietals', {
    name: DataTypes.STRING,
    description: DataTypes.TEXT,
    synonyms: {
      type: DataTypes.STRING
    },
    hero: DataTypes.STRING,
    hero_square: DataTypes.STRING,
    hero_gallery: DataTypes.STRING,
    bubble1: DataTypes.TEXT,
    bubble2:DataTypes.TEXT,
    bubble3: DataTypes.TEXT,
    bubble4:DataTypes.TEXT,
    type:{
      type: DataTypes.ENUM,
      values: ['white', 'red', 'sparkling', 'dessert']
    }
  }, {
    classMethods: {
      associate: function(models) {
        // associations can be defined here
        Varietals.belongsToMany(models.BaseAttributes, {
          through: 'VarietalsAttributes'
        });
        Varietals.belongsToMany(models.Countries, {
          through: 'CountryVarietals'
        });
        Varietals.belongsToMany(models.Foods, {
          through: 'FoodVarietals'
        });
        Varietals.belongsToMany(models.Locations, {
          through: 'VarietalsLocations'
        });
        Varietals.hasMany(models.VarietalsMedia, {
          as: 'Media',
          onDelete: 'cascade',
          hooks: true
        });

        Varietals.hasMany(models.VarietalsChats, {
          as: 'Chats',
          onDelete: 'cascade',
          hooks: true
        });

        Varietals.hasMany(models.Wines, {
          foreignKey: 'varietal_id',
          as: 'Specifics',
          onDelete: 'cascade',
          hooks: true
        });

      }
    }
  });
  return Varietals;
};
